package.json

```diff
@@ -26,24 +26,23 @@
   "main": "lib/redux.js",
   "unpkg": "dist/redux.js",
   "module": "es/redux.js",
-  "types": "types/index.d.ts",
+  "types": "lib/redux.d.ts",
```

Comment:
No, I want to call the input file `index.ts` and have the output file be `index.d.ts`, just like it is now.

Reply:
Do you realize that d.ts files are supposed to be paired (adjacent) to similarly named js files? I'm not understanding the logic here, as editors won't understand the relationship either if they don't have the same base name. Because the roll-up configuration is not writing an index.is file.

Reply:
I'm not sure I'm articulating the problem well enough, but if someone were to import `redux/redux` it wouldn't get any type information along with it because there is no redux.d.ts file.

Reply:
```
lib/index.d.ts
lib/redux.js
```
☝️ see the issue?

You are telling TypeScript users that the `index.d.ts` file relates to an `index.js` file that doesn't exist. Also, if you import directly from `lib/redux` (for some reason) you're telling TypeScript to look for a `redux.d.ts` file, which also doesn't exist. This paradigm is broken in both directions. 

The only way it does work is if you `import {} from 'redux'`, because the package.json file defines the relationship. I realize this is the most common use case, but does it not bother you that the others are completely broken?

Reply:
Gotcha. I was thinking about it wrong and also forgot about the flat bundles somehow. 

Lemme check this PR out locally and see what it actually spits out.